src/zenml/integrations/kubeflow/orchestrators/local_deployment_utils.py

```diff
@@ -203,21 +207,53 @@ def start_kfp_ui_daemon(pid_file_path: str, port: int) -> None:
         f"{port}:80",
     ]
 
-    def _daemon_function() -> None:
-        """Port-forwards the Kubeflow Pipelines UI pod."""
-        subprocess.check_call(command)
-
-    if sys.platform == "win32":
+    if not port_available(port):
+        modified_command = command.copy()
+        modified_command[-1] = "PORT:80"
+        logger.warning(
```

Comment:
How about rather than doing this we try another port (i.e. port +1) until we find an open port. Alternatively we can use an `input()` message to ask for an open port? Of course I would only do this if we are doing the `default` port not if the user has explicitly set it

Reply:
Trying different ports is definitely easy to implement if that's what we want, I'd try to avoid using `input()` in case it's run in some CI pipeline or some other automated way.

Reply:
True!